src/Features/Core/Portable/Diagnostics/DiagnosticAnalyzerService_IncrementalAnalyzer.cs

```diff
@@ -128,6 +129,7 @@ public override void RemoveProject(ProjectId projectId)
             public override bool NeedsReanalysisOnOptionChanged(object sender, OptionChangedEventArgs e)
             {
                 return e.Option.Feature == nameof(SimplificationOptions) ||
+                       e.Option.Feature == nameof(CodeStyleOptions) ||
```

Comment:
that is because IDE analyzer is using things that it shouldn't use. IDE analyzer shouldn't use roslyn option to control diagnostic engine's behavior. all these should go away once option becomes editor config thing like ruleset editor.

third party analyzer doesn't need these kind of thing since we monitor ruleset file and re-analyze if it changes. if IDE analyzer follow the pattern set for analyzers, we won't need this. but IDE analyzer did its own way so, all other parts are now need to change to let IDE analyzers works as they want to.